central/usage/datastore/datastore_impl.go

```diff
@@ -17,18 +20,39 @@ var (
 type dataStoreImpl struct {
 	clustore clustore
 	cache    cache.Cache
+	store    postgres.Store
 }
 
 var _ DataStore = (*dataStoreImpl)(nil)
 
 // Get returns the object, if it exists from the store.
-func (ds *dataStoreImpl) Get(_ context.Context, _ *types.Timestamp, _ *types.Timestamp) ([]*storage.Usage, error) {
-	return nil, errors.New("not implemented")
+func (ds *dataStoreImpl) Get(ctx context.Context, from *types.Timestamp, to *types.Timestamp) ([]*storage.Usage, error) {
+	all, err := ds.store.GetAll(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get all usage records")
+	}
+	if from == nil {
+		from, _ = types.TimestampProto(time.Time{})
+	}
+	if to == nil {
+		to = types.TimestampNow()
+	}
+	result := []*storage.Usage{}
+	for _, record := range all {
+		if record.Timestamp.Compare(from) >= 0 && record.Timestamp.Compare(to) < 0 {
+			result = append(result, record)
+		}
+	}
+	return result, nil
 }
 
 // Insert saves the current state of an object in storage.
-func (ds *dataStoreImpl) Insert(_ context.Context, _ *storage.Usage) error {
-	return errors.New("not implemented")
+func (ds *dataStoreImpl) Insert(ctx context.Context, obj *storage.Usage) error {
+	if obj.Awkward == "" {
+		obj.Awkward = uuid.NewV4().String()
```

Comment:
Can we offload this to db?

Reply:
We could with `type(serial)`, but it would become an integer ID, which we do not support.

Reply:
Using `gen_random_uuid ()` would require a non-generated schema.

Reply:
It probably wouldn't be too difficult to update the generators to handle serial and an integer ID.  We probably should do that at some point.  (I understand if that point is not now).

That said, looking at the way we use cluster_id in the cache, would it make sense to add cluster_id as the PK on the table and aggregate the results?  Would something like that impact whether we need the cache or not?  Just asking as I need to look through the rest of the PRs and get a better understanding of what we are trying to solve.